kong/router/compat.lua

```diff
@@ -282,6 +283,11 @@ end
 
 
 local function get_exp_and_priority(route)
+  if route.expression then
+    ngx_log(ngx_ERR, "expecting a traditional route while expression is given. ",
```

Comment:
`ngx.ERR` may be a little high, could we change it to `ngx.WARN`?

Reply:
It won't match any routes if this happens, and it should be handled by the administrator immediately. Thus it should be an error.